internal/tss/utils/transaction_service.go

```diff
@@ -0,0 +1,240 @@
+			},
+			IncrementSequenceNum: true,
+		},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("building transaction: %w", err)
+	}
+	tx, err = t.ChannelAccountSignatureClient.SignStellarTransaction(t.Ctx, tx, channelAccountPublicKey)
+	if err != nil {
+		return nil, fmt.Errorf("signing transaction with channel account: %w", err)
+	}
+	// wrap the transaction in a fee bump tx, signed by the distribution account
+	distributionAccountPublicKey, err := t.DistributionAccountSignatureClient.GetAccountPublicKey(t.Ctx)
+	if err != nil {
+		return nil, fmt.Errorf("getting distribution account public key: %w", err)
+	}
+
+	feeBumpTx, err := txnbuild.NewFeeBumpTransaction(
+		txnbuild.FeeBumpTransactionParams{
+			Inner:      tx,
+			FeeAccount: distributionAccountPublicKey,
+			BaseFee:    int64(t.BaseFee),
+		},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("building fee-bump transaction %w", err)
+	}
+
+	feeBumpTx, err = t.DistributionAccountSignatureClient.SignStellarFeeBumpTransaction(t.Ctx, feeBumpTx)
+	if err != nil {
+		return nil, fmt.Errorf("signing the fee bump transaction with distribution account: %w", err)
+	}
+	return feeBumpTx, nil
+}
+
+func (t *transactionService) SendTransaction(transactionXdr string) (tss.RPCSendTxResponse, error) {
+	rpcResponse, err := callRPC(t.RpcUrl, "sendTransaction", map[string]string{"transaction": transactionXdr})
+	sendTxResponse := tss.RPCSendTxResponse{}
+	sendTxResponse.TransactionXDR = transactionXdr
+	if err != nil {
+		sendTxResponse.Code.OtherCodes = tss.RPCFailCode
+		return sendTxResponse, fmt.Errorf(err.Error())
+	}
+
+	if result, ok := rpcResponse["result"].(map[string]interface{}); ok {
+		if val, exists := result["status"].(tss.RPCTXStatus); exists {
+			sendTxResponse.Status = val
+		}
+		if val, exists := result["errorResultXdr"].(string); exists {
+			sendTxResponse.Code, err = UnMarshalErrorResultXdr(val)
+		}
+		if hash, exists := result["hash"].(string); exists {
+			sendTxResponse.TransactionHash = hash
+		}
+	}
+	return sendTxResponse, err
+}
+
+func (t *transactionService) GetTransaction(transactionHash string) (tss.RPCGetIngestTxResponse, error) {
+	rpcResponse, err := callRPC(t.RpcUrl, "getTransaction", map[string]string{"hash": transactionHash})
+	if err != nil {
+		return tss.RPCGetIngestTxResponse{}, fmt.Errorf(err.Error())
+	}
+
+	getIngestTxResponse := tss.RPCGetIngestTxResponse{}
+	if result, ok := rpcResponse["result"].(map[string]interface{}); ok {
+		if status, exists := result["status"].(tss.RPCTXStatus); exists {
+			getIngestTxResponse.Status = status
+		}
+		if envelopeXdr, exists := result["envelopeXdr"].(string); exists {
+			getIngestTxResponse.EnvelopeXDR = envelopeXdr
+		}
+		if resultXdr, exists := result["resultXdr"].(string); exists {
+			getIngestTxResponse.ResultXDR = resultXdr
+		}
+		if createdAt, exists := result["createdAt"].(string); exists {
+			// we can supress erroneous createdAt errors as this is not an important field
```

Comment:
If this is not an important field could we remove it from response struct? Otherwise we should try to make sure it has a valid value.

Reply:
changed this so that we signal an error when this field cannot be parsed